src/core/Akka.Remote/Endpoint.cs

```diff
@@ -411,147 +414,174 @@ protected override void PostRestart(Exception reason)
+                    Context.System.Scheduler.ScheduleTellOnceCancelable(_settings.QuarantineSilentSystemTimeout, Self,
+                        TooLongIdle.Instance, Self);
+            Become(IdleBehavior);
         }
 
-        protected void Gated(object message)
+        private void GoToActive()
         {
-            message.Match()
-                .With<Terminated>(
-                    terminated => Context.System.Scheduler.ScheduleTellOnce(_settings.RetryGateClosedFor, Self, new Ungate(), Self))
-                .With<Ungate>(ungate =>
-                {
-                    if (_resendBuffer.NonAcked.Count > 0 || _resendBuffer.Nacked.Count > 0)
-                    {
-                        // If we talk to a system we have not talked to before (or has given up talking to in the past) stop
-                        // system delivery attempts after the specified time. This act will drop the pending system messages and gate the
-                        // remote address at the EndpointManager level stopping this actor. In case the remote system becomes reachable
-                        // again it will be immediately quarantined due to out-of-sync system message buffer and becomes quarantined.
-                        // In other words, this action is safe.
-                        if (!UidConfirmed && BailoutAt.IsOverdue)
-                        {
-                            throw new InvalidAddressAssociationException(_localAddress, _remoteAddress,
-                                new TimeoutException("Delivery of system messages timed out and they were dropped"));
-                        }
+            _maxSilenceTimer?.Cancel();
+            _maxSilenceTimer = null;
+            Become(Receiving);
+        }
 
-                        _writer = CreateWriter();
-                        //Resending will be triggered by the incoming GotUid message after the connection finished
-                        Context.Become(OnReceive);
-                    }
-                    else
-                    {
-                        Context.Become(Idle);
-                    }
-                })
-                .With<EndpointManager.Send>(send =>
+        protected void Gated(bool writerTerminated, bool earlyUngateRequested)
+        {
+            Receive<Terminated>(terminated =>
+            {
+                if (!writerTerminated)
                 {
-                    if (send.Message is ISystemMessage)
-                    {
-                        TryBuffer(send.Copy(NextSeq()));
-                    }
-                    else
+                    if (earlyUngateRequested)
+                        Self.Tell(new Ungate());
+                }
+                else
+                    Context.System.Scheduler.ScheduleTellOnce(_settings.RetryGateClosedFor, Self, new Ungate(), Self);
+                Become(() => Gated(true, earlyUngateRequested));
+            });
+            Receive<IsIdle>(idle => Sender.Tell(Idle.Instance));
```

Comment:
If we're in `Gated` behavior, then we are effectively also Idle - since nothing is being written to the transport. Reply back with `Idle`.